src/User-repo.js

```diff
@@ -2,79 +2,95 @@ class UserRepo {
     return this.makeSortedUserArray(id, dataSet)[0].date;
   };
+
   getFirstWeek(id, dataSet) {
     return this.makeSortedUserArray(id, dataSet).slice(0, 7);
   };
+
   getWeekFromDate(date, id, dataSet) {
-    let dateIndex = this.makeSortedUserArray(id, dataSet).indexOf(this.makeSortedUserArray(id, dataSet).find((sortedItem) => (sortedItem.date === date)));
-    return this.makeSortedUserArray(id, dataSet).slice(dateIndex, dateIndex + 7);
+    let user = this.makeSortedUserArray(id, dataSet).find((sortedItem) => (sortedItem.date === date));
+    let dateIndex = this.makeSortedUserArray(id, dataSet).indexOf(user);
+    let week = this.makeSortedUserArray(id, dataSet).slice(dateIndex, dateIndex + 7);
+    return week;
   };
+
   chooseWeekDataForAllUsers(dataSet, date) {
-    return dataSet.filter(function(dataItem) {
-      return (new Date(date)).setDate((new Date(date)).getDate() - 7) <= new Date(dataItem.date) && new Date(dataItem.date) <= new Date(date)
-    })
+    let allUserData = dataSet.filter((dataItem) => {
+      return (new Date(date)).setDate((new Date(date)).getDate() - 7) <= new Date(dataItem.date) && new Date(dataItem.date) <= new Date(date);
+    });
+    return allUserData;
   };
+
   chooseDayDataForAllUsers(dataSet, date) {
-    return dataSet.filter(function(dataItem) {
+    let userData = dataSet.filter((dataItem) => {
       return dataItem.date === date
     });
-  }
+    return userData;
+  };
+
   isolateUsernameAndRelevantData(dataSet, date, relevantData, listFromMethod) {
-    return listFromMethod.reduce(function(objectSoFar, dataItem) {
+    let userData = listFromMethod.reduce((objectSoFar, dataItem) => {
       if (!objectSoFar[dataItem.userID]) {
-        objectSoFar[dataItem.userID] = [dataItem[relevantData]]
+        objectSoFar[dataItem.userID] = [dataItem[relevantData]];
       } else {
-        objectSoFar[dataItem.userID].push(dataItem[relevantData])
+        objectSoFar[dataItem.userID].push(dataItem[relevantData]);
       }
       return objectSoFar;
     }, {});
-  }
+    return userData;
+  };
+
   rankUserIDsbyRelevantDataValue(dataSet, date, relevantData, listFromMethod) {
-    let sortedObjectKeys = this.isolateUsernameAndRelevantData(dataSet, date, relevantData, listFromMethod)
-    return Object.keys(sortedObjectKeys).sort(function(b, a) {
-      return (sortedObjectKeys[a].reduce(function(sumSoFar, sleepQualityValue) {
-        sumSoFar += sleepQualityValue
-        return sumSoFar;
-      }, 0) / sortedObjectKeys[a].length) - (sortedObjectKeys[b].reduce(function(sumSoFar, sleepQualityValue) {
-        sumSoFar += sleepQualityValue
-        return sumSoFar;
-      }, 0) / sortedObjectKeys[b].length)
+    let sortedObjectKeys = this.isolateUsernameAndRelevantData(dataSet, date, relevantData, listFromMethod);
+    let dataValue = Object.keys(sortedObjectKeys).sort((b, a) => {
+      return (this.getAverageRelevantData(sortedObjectKeys[a])) - (this.getAverageRelevantData(sortedObjectKeys[b]));
```

Comment:
Awesome refactor on this one Bryce! SO much better to have that dynamic helper method too.